src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -129,22 +148,41 @@ def createPileup(self, pdict, rseList):
             return [err.error()]
         return []
 
-    def updatePileup(self, doc, rseList=None, validate=True):
+    def updatePileup(self, doc, rseList=None, validate=True, userDN=''):
         """
         Update pileup data with provided input
 
         :param doc: a dictionary of pieleup data to be updated
         :param rseList: a list of valid RSE names
         :param validate: boolean defining whether the doc needs
                          to be validated or not
+        :param userDN: a string, user DN used in HTTP PUT request
         :return: list of MSPileupError or empty list
         """
+        self.logger.info("Updating pileup document %s", doc)
         rseList = rseList or []
         pname = doc.get('customName', '')
         spec = {'customName': pname}
         if pname == '':
             pname = doc.get('pileupName', '')
             spec = {'pileupName': pname}
+
+        # add transition record if doc contains new fraction
+        fraction = doc.get('containerFraction', 0)
+        transition = doc.get('transition', [])
+        if fraction > 0:
+            # check previous fraction value and add new transition record if it is
+            # differ from current one or does not exist
+            transition = doc.get('transition', [])
+            if not transition or transition[-1]['containerFraction'] != fraction:
+                transitionRecord = {'containerFraction': fraction,
```

Comment:
Valentin, on this containerFraction provided by the user. Should we define a given scale to this number (digits to the right of the decimal point) and avoid any float precision issues with Python? If so, I would stick to scale 2 (i.e. range of 0.00, 0.01, ... 0.99, 1.00)

Reply:
please do not complicate things. Python correctly handle precision to my knowledge while it may print it not correctly though. I'm not in favor of introducing another complexity where it is not required.

Reply:
I never liked real numbers in python due to some unpredictable behaviors seen, hence my suggestion to actually avoid complications in the future. Nonetheless, I have no strong opinion here and I am fine as is.